app/services/S3.scala

```diff
@@ -44,116 +60,111 @@ object S3 extends S3Client with StrictLogging {
     .withCredentials(Aws.credentialsProvider)
     .build()
 
-  def get(objectSettings: S3ObjectSettings)(implicit ec: ExecutionContext): Future[Either[String,RawVersionedS3Data]] = Future {
-    try {
-      val s3Object = s3Client.getObject(objectSettings.bucket, objectSettings.key)
-
-      val version = s3Object.getObjectMetadata.getVersionId
-
-      val stream = s3Object.getObjectContent
-      val raw: String = try {
-        scala.io.Source.fromInputStream(stream).mkString
-      } finally {
-        stream.close()
-        s3Object.close()
+  def get: S3Action[RawVersionedS3Data] = { objectSettings =>
+    val s3ObjectAndStream = for {
+      s3Object <- ZManaged.fromAutoCloseable(effectBlocking(s3Client.getObject(objectSettings.bucket, objectSettings.key)))
```

Comment:
`ZManaged.fromAutoCloseable` ensures that `.close()` is called on the managed resource